flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -944,6 +955,15 @@ class PFTDumper {
     outputStream << "EndContains\nEndModuleLike\n\n";
   }
 
+  // Top level directives
+  void dumpCompilerDirectiveUnit(
+      llvm::raw_ostream &outputStream,
+      const lower::pft::CompilerDirectiveUnit &directive) {
+    outputStream << getNodeIndex(directive) << " ";
+    outputStream << "CompilerDirective: ";
```

Comment:
Does a directive have some text representation available from the front-end? (Not sure `AsFortran` makes sense, but...)

Reply:
There is no `AsFortran` for `CompilerDirective`. Like with many nodes, `lib/Parser/unparse.cpp` has code to dump the directive, but it is not accessible, however, we can use easily dump the source text, so I have added this.